orderbook/src/api/cancel_order.rs

```diff
@@ -1 +1,75 @@
-// TODO - write me.
\ No newline at end of file
+use crate::orderbook::{Orderbook, OrderCancellationResult};
+use anyhow::Result;
+use model::order::OrderCancellation;
+use std::{convert::Infallible, sync::Arc};
+use warp::{hyper::StatusCode, Filter, Rejection, Reply};
+
+const MAX_JSON_BODY_PAYLOAD: u64 = 1024 * 16;
+
+fn extract_cancellation() -> impl Filter<Extract = (OrderCancellation,), Error = Rejection> + Clone {
+    // (rejecting huge payloads)...
+    warp::body::content_length_limit(MAX_JSON_BODY_PAYLOAD).and(warp::body::json())
+}
+
+pub fn cancel_order_request() -> impl Filter<Extract = (OrderCancellation,), Error = Rejection> + Clone
+{
+    warp::path!("orders")
+        .and(warp::delete())
+        .and(extract_cancellation())
+}
+
+pub fn cancel_order_response(result: Result<OrderCancellationResult>) -> impl Reply {
+    let (body, status_code) = match result {
+        Ok(OrderCancellationResult::Cancelled) => (warp::reply::json(&"Cancelled"), StatusCode::ACCEPTED),
```

Comment:
According to https://en.wikipedia.org/wiki/List_of_HTTP_status_codes

> The request has been accepted for processing, but the processing has not been completed. The request might or might not be eventually acted upon, and may be disallowed when processing occurs.

which I'm not sure is exactly true in this case.

I don't care too much about the actual code we use, but given our OpenAPI spec says we return 200 in this case  let's maybe do as documented?